tutorials/ExploringDeutschJozsaAlgorithm/DeutschJozsaAlgorithmTutorial_P2.ipynb

```diff
@@ -0,0 +1,313 @@
+    "\n",
+    "> **Note:**\n",
+    "All code snippets before Exercise 3 are just examples. They don't need to be modified and are not covered by tests."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### $f(x) \\equiv 0$\n",
+    "\n",
+    "This is the easiest function to implement: if $f(x) \\equiv 0$, $U_f |x\\rangle \\equiv (-1)^0 |x\\rangle = |x\\rangle$. \n",
+    "This means that $U_f$ is an identity - a transformation which does absolutely nothing! \n",
+    "This is very easy to express in Q#:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "operation PhaseOracle_Zero (x : Qubit) : Unit {\n",
+    "    // Do nothing...\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### $f(x) \\equiv 1$\n",
+    "\n",
+    "The second constant function is slightly trickier: if $f(x) \\equiv 1$, $U_f |x\\rangle \\equiv (-1)^1 |x\\rangle = - |x\\rangle$. \n",
+    "Now $U_f$ is a negative identity, i.e., a transformation which applies a global phase of $-1$ to the state. \n",
+    "A lot of algorithms just ignore the global phase accumulated in them, since it is not observable. \n",
+    "However, if we want to be really meticulous, we can use Q# library operation [Microsoft.Quantum.Intrinsic.R](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.intrinsic.r) which performs a given rotation around the given axis. \n",
+    "When called with `PauliI` axis, this operation applies a global phase; since it doesn't take the input into account, it can be applied to any qubit, for example, the first qubit of the input."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "// Open namespace where the library function PI() is defined\n",
+    "open Microsoft.Quantum.Math;\n",
+    "\n",
+    "operation PhaseOracle_One (x : Qubit) : Unit {\n",
+    "    // Apply a global phase of -1\n",
+    "    R(PauliI, 2.0 * PI(), x);\n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### $f(x) = x$\n",
+    "\n",
+    "$$U_f |x\\rangle = (-1)^{f(x)} |x\\rangle = (-1)^{x} |x\\rangle$$\n",
+    "\n",
+    "This means that we need to do nothing if the qubit is in the $|0\\rangle$ state, and apply a phase of $-1$ if it is in the $|1\\rangle$ state. This is exactly the effect of the [Z gate](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.intrinsic.z); as a reminder, \n",
+    "\n",
+    "$$Z = \\begin{bmatrix} 1 & 0 \\\\ 0 & -1\\end{bmatrix}: Z |0\\rangle = |0\\rangle, Z |1\\rangle = -|1\\rangle$$"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "operation PhaseOracle_X (x : Qubit) : Unit {\n",
```

Comment:
It's slightly confusing that this is called `PhaseOracle_X`, maybe it can be called `PhaseOracle_Negate` or `_Invert`?

Reply:
I thought about it, but in this batch of oracles they are called for their classical functions - Zero instead of Identity and One instead of GlobalPhase, so I think X aligns better with this